src/core/qgsaction.h

```diff
@@ -119,6 +119,48 @@ class CORE_EXPORT QgsAction
     {}
 #endif
 
+    /**
+     * Create a new QgsAction
+     *
+     * \param id                   The unique identifier of this action
+     * \param type                 The type of this action
+     * \param description          A human readable description string
+     * \param action               The action text. Its interpretation depends on the type
+     * \param icon                 Path to an icon for this action
+     * \param capture              If this is set to TRUE, the output will be captured when an action is run
+     * \param shortTitle           A short string used to label user interface elements like buttons
+     * \param actionScopes         A set of scopes in which this action will be available
+     * \param notificationMessage  A particular message which reception will trigger the action
+     * \param enabledOnlyWhenEditable if TRUE then action is only enable in editmode. Not available in Python bindings.
+     */
+#ifndef SIP_RUN
+    QgsAction( const QUuid &id, ActionType type, const QString &description, const QString &action, const QString &icon, bool capture, const QString &shortTitle = QString(), const QSet<QString> &actionScopes = QSet<QString>(), const QString &notificationMessage = QString(), bool enabledOnlyWhenEditable = false )
+      : mType( type )
+      , mDescription( description )
+      , mShortTitle( shortTitle )
+      , mIcon( icon )
+      , mCommand( action )
+      , mCaptureOutput( capture )
+      , mActionScopes( actionScopes )
+      , mNotificationMessage( notificationMessage )
+      , mId( id )
+      , mIsEnabledOnlyWhenEditable( enabledOnlyWhenEditable )
+    {}
+#else
+    QgsAction( const QUuid &id, ActionType type, const QString &description, const QString &action, const QString &icon, bool capture, const QString &shortTitle = QString(), const QSet<QString> &actionScopes = QSet<QString>(), const QString &notificationMessage = QString() )
+      : mType( type )
+      , mDescription( description )
+      , mShortTitle( shortTitle )
+      , mIcon( icon )
+      , mCommand( action )
+      , mCaptureOutput( capture )
+      , mActionScopes( actionScopes )
+      , mNotificationMessage( notificationMessage )
+      , mId( id )
+      , mIsEnabledOnlyWhenEditable( enabledOnlyWhenEditable )
+    {}
+#endif
```

Comment:
```suggestion
    QgsAction( const QUuid &id, ActionType type, const QString &description, const QString &action, const QString &icon, bool capture, const QString &shortTitle = QString(), const QSet<QString> &actionScopes = QSet<QString>(), const QString &notificationMessage = QString(), bool enabledOnlyWhenEditable SIP_PYARGREMOVE = false )
      : mType( type )
      , mDescription( description )
      , mShortTitle( shortTitle )
      , mIcon( icon )
      , mCommand( action )
      , mCaptureOutput( capture )
      , mActionScopes( actionScopes )
      , mNotificationMessage( notificationMessage )
      , mId( id )
      , mIsEnabledOnlyWhenEditable( enabledOnlyWhenEditable )
    {}
```

can you try this with SIP_PYARGREMOVE?
if this doesn't work, you can at least remove the init code from the SIP_RUN part as it is useless (see that you left a non-inited variable `enabledOnlyWhenEditable`)

you can apply the same correction to the other constructor.

Reply:
> can you try this with SIP_PYARGREMOVE?
> if this doesn't work, you can at least remove the init code from the SIP_RUN part as it is useless (see that you left a non-inited variable `enabledOnlyWhenEditable`)
> 
> you can apply the same correction to the other constructor.

Yeah, I just copied the already existing ctors, I also wondered why they were like that.